src/gui/options_imp.cpp

```diff
@@ -776,6 +786,7 @@ void options_imp::loadOptions()
         checkDownloadLimit->setChecked(true);
         spinDownloadLimit->setEnabled(true);
         spinDownloadLimit->setValue(intValue);
+        setLimitStep(spinDownloadLimit, intValue);
```

Comment:
This is needed to so that the step size is set correctly when the program starts. Without it, the first rate limit change will always have step size 1, and then it's adjusted when setLimitStep() is called.

I added calls to the adapt function for the disabled cases too, so that the grey-out defaults (100 KiB/s for the global rate limits) will have the correct step sizes right away when enabled.